assets/src/web-stories-block/block/edit.js

```diff
@@ -45,7 +45,13 @@ import { icon } from './';
 const { config } = webStoriesBlockSettings;
 
 function WebStoriesEdit({ attributes, setAttributes, className, isSelected }) {
-  const { blockType, viewType } = attributes;
+  const { blockType, viewType, url } = attributes;
+
+  if (!blockType && url && url.length) {
+    setAttributes({
+      blockType: BLOCK_TYPE_URL,
+    });
+  }
```

Comment:
Would something like this work?

```js
// If block type is not set, but there is a URL,
// this block was likely migrated from its previous form.
if (!blockType && url?.length) {
  return (
    <ConfigProvider config={config}>
      <StoryEmbedEdit
        icon={icon}
        attributes={attributes}
        setAttributes={setAttributes}
        className={className}
        isSelected={isSelected}
      />
    </ConfigProvider>
  );
}
```

Maybe together with an `useEffect` to set the blockType?

```js
useEffect(() => {
  if (!blockType && url?.length) {
    setAttributes({
      blockType: BLOCK_TYPE_URL,
    });
  }
}, [blockType, url, setAttributes]);
```

Reply:
I see you added the `useEffect` part.

Does this still cause this "flash" when first loading the block? Does the above `if (!blockType && url?.length) { return ( .... ) }` prevent this?

Reply:
The flash is still there. I'm looking into the docks and Gutenberg source if I find anything will update it. 
Also, I did remove Embed block files and moved the required ones to the new block directory.

Reply:
I thought with my `if (!blockType && url?.length) { return ( ... ) }` suggestion above this flash wouldn't be happening. Have you tried it?

Reply:
Update the deprecation definition and now it works, apparently, I needed to add `isEligible` for the migrate to work.

https://github.com/rtCamp/web-stories-wp/blob/5b21364cceb58fdf12470502397d1519298ca31d/assets/src/web-stories-block/block/deprecated.js#L110

and removed that `useEffect()` which was manually setting the `blockType`.

Reply:
Nice!!